cluster/examples/kubernetes/cassandra/operator.yaml

```diff
@@ -0,0 +1,227 @@
+
+# Namespace where Cassandra Operator will live
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook-cassandra-system
+
+---
+
+# Cassandra Cluster CRD
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: clusters.cassandra.rook.io
+spec:
+  group: cassandra.rook.io
+  names:
+    kind: Cluster
+    listKind: ClusterList
+    plural: clusters
+    singular: cluster
+  scope: Namespaced
+  version: v1alpha1
+  validation:
+    openAPIV3Schema:
```

Comment:
Being able to automatically generate this CRD schema validation, always keeping it in sync with types.go, would be very beneficial.  I would love to see some of the ecosystem's more powerful operator/controller tools get integrated with Rook, as captured in https://github.com/rook/rook/issues/1981#issuecomment-421058734.

that is definitely out of scope for this PR, but you probably felt the pain of having to keep this validation up to date by hand so far :(